src/Microsoft.Azure.WebJobs.Host/Config/ExtensionConfigContext.cs

```diff
@@ -63,12 +64,19 @@ public Uri GetWebhookHandler()
         /// <returns></returns>
         public FluentBindingRule<TAttribute> AddBindingRule<TAttribute>() where TAttribute : Attribute
         {
-            bool hasBindingAttr = typeof(TAttribute).GetCustomAttributes(typeof(BindingAttribute), false).Length > 0;
-            if (!hasBindingAttr)
+            var bindingAttrs = typeof(TAttribute).GetCustomAttributes(typeof(BindingAttribute), false);
+            if (!bindingAttrs.Any())
             {                
                 throw new InvalidOperationException($"Can't add a binding rule for '{typeof(TAttribute).Name}' since it is missing the a {typeof(BindingAttribute).Name}");
             }
 
+            var isTrigger = typeof(TAttribute).Name.EndsWith("TriggerAttribute", StringComparison.OrdinalIgnoreCase);
+            var bindingAttr = bindingAttrs.First() as BindingAttribute;
```

Comment:
>as BindingAttribute; [](start = 51, length = 20)

I think you can remove this since we already did the type check on line 67.   Just do a direct cast. That will also simplify line 75